src/plugins/expressions/common/expression_functions/specs/ui_setting.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionDefinition } from 'src/plugins/expressions/common';
+import { getUiSettingsFactory } from '../../../common';
+import { UiSetting } from '../../expression_types/specs/ui_setting';
+
+export interface UiSettingArguments {
+  default?: unknown;
+  parameter: string;
+}
+
+export type ExpressionFunctionUiSetting = ExpressionFunctionDefinition<
+  'uiSetting',
+  unknown,
+  UiSettingArguments,
+  Promise<UiSetting>
+>;
+
+export const uiSetting: ExpressionFunctionUiSetting = {
+  name: 'uiSetting',
+  help: i18n.translate('expressions.functions.uiSetting.help', {
+    defaultMessage: 'Returns a UI settings parameter value.',
+  }),
+  args: {
+    default: {
+      help: i18n.translate('expressions.functions.uiSetting.args.default', {
+        defaultMessage: 'A default value in case of the parameter is not set.',
+      }),
+    },
+    parameter: {
+      aliases: ['_'],
+      help: i18n.translate('expressions.functions.uiSetting.args.parameter', {
+        defaultMessage: 'The parameter name.',
+      }),
+      required: true,
+      types: ['string'],
+    },
+  },
+  async fn(input, { default: defaultValue, parameter }, { getKibanaRequest }) {
+    const uiSettingsFactory = getUiSettingsFactory();
```

Comment:
rather than using a factory, wrap this definition in a function and pass a method in that provided an optional getKibanaRequests returns `uiSettingsClient`. take a look at `plugins/search/common/search/expressions/esdsl`.

Reply:
I think we should try to keep the same pattern, rather than having it done a bit different in every place.

Reply:
I reworked the way we register all the functions in the expressions plugin. The `uiSetting` follows the same pattern and has two adapters, one for the server and another one for the public.